### ISSUE TITLE
Disable reflection for extensions

### ISSUE THREAD

**Issue by AndrolGenhald** [vimeo/psalm#7515]
We now load extension stubs based on composer.json and xml configuration instead of using whichever extensions are currently loaded, but I believe Psalm will still use reflection to check extension types. We should make sure reflection isn't used for extensions so that when an extension is disabled Psalm doesn't fail to find errors due to its usage.

**Comment by orklah** [comment]
Well, if the extension is available, we should try to Reflection it. But yeah, if it's not available at all, a message suggesting to use https://github.com/vimeo/psalm/issues/7513#issuecomment-1024765248 to contribute to Psalm or enabling the extension would be cool

**Comment by weirdan** [comment]
> Well, if the extension is available, we should try to Reflection it.

It might be available but still be disabled.

**Comment by AndrolGenhald** [comment]
> > Well, if the extension is available, we should try to Reflection it.
> 
> It might be available but still be disabled.

Yeah, that.

Say you have the xml extension installed locally (you have to to be able to run Psalm), but your composer.json doesn't require it and some of your deployment environments don't have it. It would be good if Psalm generated issues for xml functions not existing, since they might not exist in other environments where the code is deployed.

**Comment by orklah** [comment]
Yeah, adding a message warning to say it's not enabled is good, but yeah, it would be weird not to use the perfectly good code we already have to generate a better analysis anyway if the extension is here
